### ISSUE TITLE
TTree.array does not respect entrystop parameter when reading incomplete file

### ISSUE THREAD

**Issue by bersbersbers** [scikit-hep/uproot#87]
I have a broken ROOT file from which ROOT is able to recover 6000 entries.

Opening in uproot and reading an array yields the error
`IndexError, indexes 9812:28799 are beyond the end of data source 'Coincidences.root'`

This is quite understable, but I had expected entrystop to be able to read only the first 1000 entries:
````
import uproot
RootFile = uproot.open("Coincidences.root")["Coincidences"].array("runID", entrystop=1000)
````

Yet it yields the same error.

**Comment by jpivarski** [comment]
Let me know if you want uproot to support partial-file recovery by reopening this issue. I'll consider it a feature request. Thanks!

**Comment by bersbersbers** [comment]
Thank you for these explanations! I was under the impression that a root file can easily be read partially. If that is not the case, this is indeed only a feature request, and a not too urgent one.

Rest assured that I do my research based on complete files ;) I usually use partial files to track the process of long-running (days to weeks) simulations - I simply copy the files that are just being written to a different location, and root it able to recover them just fine. I have yet to find out how to make root save a recovered file as a new, clean file - this would make this issue irrelevant. But for the time being, I am OK with running a short simulation first and then hope that only changing a single parameter will not destroy everything ;)

**Comment by jpivarski** [comment]
Okay, good to know that it isn't urgent and your use-case makes a lot of sense, now that I think about it.

If you want me to investigate it as a feature request, point me to a copy of such a file and I'll see what I can make of it. I can't guarantee a general solution because I don't know if they'd all be broken in (qualitatively) the same way.

You might be interested in this: https://indico.cern.ch/event/658060/contributions/2898569/ I gave a talk (more of a discussion) about the different general features that a file format can have or not have. ROOT is not in the "streamable" category, but others are. Maybe you'd want to monitor your jobs by outputting in two forms: archival ROOT and something streamable, like raw binary (just write the numerical array data directly to a file, with no intermediary— no need for a library). The raw binary is not self-documenting, which is bad for archiving, but as long as you remember how to interpret it, you can read it at _any_ stage of filling. Then you throw it away when your job is done.

P.S. Raw binary files can be read by Numpy's fromstring or fromfile commands. You may need to read it as dtype "uint8" and truncate its size to a multiple of your row width before viewing as the correct type. If you want an example, I'll send one by e-mail.

**Comment by bersbersbers** [comment]
Thanks! If a come across another one of these files with a similar error I let you know. I have somewhat of a hard time reproducing that issue, since it was some effort to at least _open_ the file: before opening it in uproot, I had to open in ROOT's TBrowser and "Save as" from there.  This is how I created that file that gave me the above error, but I don't have it any more.

What I can tell from looking at other files (the ones that don't even open), they seem to be written every 100 KB or so, and the branch name is associated with a number that is increasing, such as "Coincidences;5", "Coincidences;6", etc - and ROOT seems to be able to find a/the last working one. I could upload one of those next week if you're interested (as I said, there is no urgent need for me).
